UKHO.D365CallbackDistributorStub.API/UKHO.D365CallbackDistributorStub.API/Services/CallbackService.cs

```diff
@@ -1,21 +1,33 @@
 ﻿using System.Diagnostics.CodeAnalysis;
+using System.Net;
 using UKHO.D365CallbackDistributorStub.API.Models.Request;
 
 namespace UKHO.D365CallbackDistributorStub.API.Services
 {
     [ExcludeFromCodeCoverage]
-    public class CallbackService
+    public class CallbackService 
     {
         private static readonly Queue<RecordCallbackRequest> s_recordCallbackRequestQueue = new();
-        public static bool SaveCallbackRequest(CallbackRequest callbackRequest, string subscriptionId)
+        private static readonly List<CommandCallbackRequest> s_commandCallbackRequestList = new();
+        private const HttpStatusCode _noContent = HttpStatusCode.NoContent;
+        private readonly ILogger<CallbackService> _logger;
+
+        public CallbackService(ILogger<CallbackService> logger)
+        {
+            _logger = logger;
+        }
+
+        public static bool SaveCallbackRequest(CallbackRequest callbackRequest, string subscriptionId, HttpStatusCode? httpStatusCode)
         {
             try
             {
                 s_recordCallbackRequestQueue.Enqueue(new RecordCallbackRequest
                 {
                     CallBackRequest = callbackRequest,
                     Guid = Guid.NewGuid(),
-                    SubscriptionId = subscriptionId
+                    SubscriptionId = subscriptionId,
+                    TimeStamp = CommonService.ToRfc3339String(DateTime.UtcNow),
+                    HttpStatusCode = httpStatusCode ?? _noContent
```

Comment:
Change this as per above change:
```suggestion
                    HttpStatusCode = httpStatusCode ?? NoContent
```

Reply:
Changes done